File: python/백준/정렬/1461 도서관.py
N, M = map(int, input().split())

books = list(map(int, input().split()))


left = []
right = []

for x in books:
    if x < 0:
        left.append(abs(x))
    else:
        right.append(x)

left.sort(reverse=True)
right.sort(reverse=True)

result = 0

for i in range(0, len(left), M):
    result += left[i] * 2

for i in range(0, len(right), M):
    result += right[i] * 2

# 🔥 가장 먼 거리 한 번 빼기
max_dist = 0
if left:
    max_dist = max(max_dist, left[0])
if right:
    max_dist = max(max_dist, right[0])

result -= max_dist

print(result)

File: cpp/백준/정렬/1461 도서관.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <functional>
#include <cmath>
using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);

    int N, M;
    cin >> N >> M;

    vector<int> left, right;
    for (int i = 0; i < N; ++i) {
        int x;
        cin >> x;
        if (x < 0) left.push_back(abs(x));
        else right.push_back(x);
    }

    sort(left.begin(), left.end(), greater<int>());
    sort(right.begin(), right.end(), greater<int>());

    long long result = 0;
    for (size_t i = 0; i < left.size(); i += M) {
        result += (long long)left[i] * 2;
    }
    for (size_t i = 0; i < right.size(); i += M) {
        result += (long long)right[i] * 2;
    }

    int max_dist = 0;
    if (!left.empty()) max_dist = max(max_dist, left[0]);
    if (!right.empty()) max_dist = max(max_dist, right[0]);

    result -= max_dist;

    cout << result << "\n";
    return 0;
}
